slam/alarm_tree_view.py

```diff
@@ -84,6 +90,66 @@ def __init__(
 
         self.layout.addWidget(self.tree_view)
 
+    def handleThresholdDisplay(self):
+        indices = self.tree_view.selectedIndexes()
+        index = indices[0]
+        alarm_item = self.treeModel.getItem(index)
+        hihi = high = low = lolo = -1
+
+        # If not a leaf its an invalid 'cainfo' call which could stall things for a while.
+        if not alarm_item.is_leaf():
+            # Don't display any of the threshold-display actions if selected non-leaf
+            self.display_thresholds_menu.clear()
+            return
+
+        # Avoid calling 'cainfo' on undefined alarm since causes the call to stall for a bit.
+        # Also we don't want thresholds from an undefined alarm anyway.
+        if alarm_item.is_undefined_or_invalid():
+            # Don't display any of the threshold-display actions if alarm-item undefined
+            self.display_thresholds_menu.clear()
+            return
+
+        # Make pv_object if first time item's threshold is requested
+        if alarm_item.pv_object is None:
+            # Update the values only when user requests them in right-click menu
+            alarm_item.pv_object = PV(alarm_item.name, auto_monitor=False)
+
+        # Do a get call we can quickly timeout, so if PV not-connected don't
+        # need to wait for slower get_ctrlvars() call.
+        # 0.1 is small arbitrary value, can be made larger if timing-out for
+        # actually connected PVs.
+        if alarm_item.pv_object.get(timeout=0.1) is None:
+            return
+        alarm_item_metadata = alarm_item.pv_object.get_ctrlvars()
+
+        # Getting data can fail for some PV's, good metadata will always have a key for all 4 limits (nan if not set),
+        # in this case don't display any threshold sub-menus
+        if alarm_item_metadata is None:
+            logger.debug(f"Can't connect to PV: {alarm_item.name}")
```

Comment:
```suggestion
            logger.warn(f"Can't connect to PV: {alarm_item.name}")
```
Only last thing I'd change, for logging messages that are useful to the user I think at least warn is more useful since most users won't run with logging set to debug by default